include/mqtt/property.hpp

```diff
@@ -168,17 +159,8 @@ struct binary_property {
 };
 
 struct string_property : binary_property {
-    template <
-        typename Buffer,
-        typename = std::enable_if_t<
-            !std::is_same<
-                std::decay_t<Buffer>,
-                string_property
-            >::value
-        >
-    >
-    string_property(property::id id, Buffer&& buf, bool already_checked)
-        :binary_property(id, std::forward<Buffer>(buf)) {
+    string_property(property::id id, buffer buf, bool already_checked)
```

Comment:
Should the already_checked parameter be defaulted to false?

Reply:
I don't think so. `detail::string_property` is only for internal use. And all properties that use `detail::string_property` pass to `already_checked explicitly.

Reply:
ok